.github/workflows/main.yml

```diff
@@ -50,26 +108,47 @@ jobs:
           fetch-depth: 50
           token: ${{ secrets.GITHUB_TOKEN }}
 
-      - if: steps.step_conditionals_handler.outputs.is_push_to_master == 'true'
+      - if: matrix.release
+        name: Check latest tag
+        id: check_latest_tag
+        shell: pwsh
+        run: |
+          $LATEST_TAG = git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags "https://github.com/$env:GIT_REPOSITORY.git" '*.*.*' | tail --lines=1 | cut --delimiter='/' --fields=3
+          $LATEST_VERSION = [System.Version]::Parse($LATEST_TAG -replace 'v')
+          echo "::set-output name=semver::$(echo "$($LATEST_VERSION.Major).$($LATEST_VERSION.Minor).$($LATEST_VERSION.Build)")"
+        env:
+          GIT_REPOSITORY: ${{ github.repository }}
+
+      - if: matrix.release && steps.check_latest_tag.outputs.semver != ''
         name: Bump GitHub tag
         id: tag_generator
-        uses: mathieudutour/github-tag-action@v5
+        uses: soumyamahunt/github-tag-action@test-other-ver-support
```

Comment:
Is this your own version of tag action?

Reply:
I was about to comment about this, I opened a [PR](https://github.com/mathieudutour/github-tag-action/pull/73) in the action repo that will allow notepads to use the tag action without having to follow semver style tags. Currently I am keeping this for testing, but if the PR accepted we will follow the main repo.

Reply:
> @JasonStein I have some suggestion about release strategy. Right now any push commit with the proper format triggers a release, the drawback of this is the changelog will only contain that specific commit and you will have to later modify changelog. What I want to propose is trigger release with `workflow_dispatch` event with an [input](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#example-workflow-configuration), this way we can always follow the format for all commits and the changelog will be generated from appropriate commits without requiring you to modify anything later.

100% agreed.